open_flamingo/eval/evaluate.py

```diff
@@ -509,14 +492,14 @@ def evaluate_captioning(
 
 
 def evaluate_vqa(
-    args,
-    eval_model,
-    seed=42,
-    max_generation_length=5,
-    num_beams=3,
-    length_penalty=-2.0,
-    num_shots=8,
-    vqa_dataset="vqav2",
+    args: argparse.Namespace,
```

Comment:
This is a personal preference (that is strongly violated in other parts of the codebase already) but I think it is much more transparent to explicitly include the individual parameters that will be accessed from args here (and then document them individually), instead of passing the entire arguments object. It also makes this function almost impossible to use externally if it requires an argparse object instead of a set of parameters.

I would suggest adding `train_image_dir_path`, `train_questions_json_path`, etc as parameters and not passing `args` at all.

If the problem here is that the user needs to choose either okvqa or vqav2 (but not both), then maybe we need to tweak the logic of the script above (i.e. take separate args for the train image for both (e.g. `okvqa_train_image_dir_path` and `vqav2train_image_dir_path`) and call this function separately for each.

Reply:
Ok since this is a bit out of scope for this PR (and will likely need to change in many other places) I will merge this in and we can fix this in another PR.